classes/local/client_factory.php

```diff
@@ -44,10 +44,17 @@ class client_factory {
     public static function get_client(string $class, array $opts): AwsClient {
         // Modify the opts to add HTTP timeouts.
         if (empty($opts['http'])) {
-            $opts['http'] = ['connect_timeout' => HOURSECS];
-        } else if (is_array($opts['http'] && !array_key_exists('connect_timeout', $opts['http']))) {
-            // Try not to override existing settings.
-            $opts['http']['connect_timeout'] = HOURSECS;
+            $opts['http'] = ['connect_timeout' => 5, 'timeout' => HOURSECS];
+
+        } else if (is_array($opts['http'])) {
+            if (!array_key_exists('connect_timeout', $opts['http'])) {
+                // Try not to override existing settings.
+                $opts['http']['connect_timeout'] = 5;
+            }
+            if (!array_key_exists('timeout', $opts['http'])) {
+                // Try not to override existing settings.
+                $opts['http']['timeout'] = HOURSECS;
```

Comment:
I think this is good as is, the only thing we could add is check $opts first, then get_config and then finally HOURSECS which gives us some room to tweak this further case by case down the road